### PR TITLE
Support fetching staging slot URL in end-to-end test project

### PR DESCRIPTION
**Pull request by joelverhagen**
This mimics the script used in functional tests to allow running end-to-end tests against the gallery staging slot URL.

The parameters values live in Octopus and are passed to a VSTS build.

### CODE REVIEW

File: src/NuGet.Services.EndToEnd/EnvironmentSettings.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace NuGet.Services.EndToEnd
+{
+    public static class EnvironmentSettings
```
Comment (skofman1) [inline src/NuGet.Services.EndToEnd/EnvironmentSettings.cs]: if there is any code duplicated from FunctionalTests, please add a comment on the class, so we will know what needs to be consolidated

File: src/NuGet.Services.EndToEnd/BasicTests.cs [hunk 2]
```diff
@@ -8,16 +8,17 @@
 
 namespace NuGet.Services.EndToEnd
 {
-    public class BasicTests
+    public class BasicTests : IClassFixture<TrustedHttpsCertificatesFixture>
```
Comment (skofman1) [inline src/NuGet.Services.EndToEnd/BasicTests.cs]: This class needs a less general name. When we add more tests, we will need clearity on where to put them..

File: src/NuGet.Services.EndToEnd/EnvironmentSettings.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        };
+
+        public static string GalleryBaseUrl => GetEnvironmentVariable("GalleryBaseUrl", required: true);
+
+        public static string[] TrustedHttpsCertificates
+        {
+            get
+            {
+                var value = GetEnvironmentVariable("TrustedHttpsCertificates", required: false);
+                if (value == null)
```
Comment (scottbommarito) [inline src/NuGet.Services.EndToEnd/EnvironmentSettings.cs]:
```
var value = GetEnvironmentVariable("TrustedHttpsCertificates", required: false) ?? new string[0];
```

Comment (scottbommarito) [inline src/NuGet.Services.EndToEnd/EnvironmentSettings.cs]: Also I would probably name this variable something else--naming it `value` like a setter makes this slightly confusing.

File: src/NuGet.Services.EndToEnd/BasicTests.cs (hunk 2 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/BasicTests.cs]: This is essentially a place-holder until we have real tests.

File: src/NuGet.Services.EndToEnd/EnvironmentSettings.cs (hunk 3 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/EnvironmentSettings.cs]: Variable name -- sure.

Cannot coalesce `string` to `string[]`.

File: src/NuGet.Services.EndToEnd/EnvironmentSettings.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/EnvironmentSettings.cs]: Done.

File: src/NuGet.Services.EndToEnd/EnvironmentSettings.cs (hunk 3 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/EnvironmentSettings.cs]: Done.

**Pull request merged**